### PR TITLE
App 781 from scratch

### PR DESCRIPTION
**Pull request by pdarde-daitan**
Change all the configurator app architecture by using redux-saga.

### CODE REVIEW

File: integration-fe/src/js/app.jsx [hunk 1]
```diff
@@ -1,18 +1,12 @@
-/* eslint-disable no-unused-vars */
+import 'babel-polyfill';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/js/app.jsx]: Do you really need this?

File: integration-fe/src/sagas/getAppName.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-debugger */
+import { call, put } from 'redux-saga/effects';
+import { getAppName as appName } from './apiCalls';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/sagas/getAppName.js:3]: Why are you doing that?

File: integration-fe/src/sagas/getBaseWebHookURL.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-debugger */
+import { call, put } from 'redux-saga/effects';
+import { getBaseWebHookURL as getURL } from './apiCalls';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/sagas/getBaseWebHookURL.js:3]: Why are you doing that?

File: integration-fe/src/sagas/getInstance.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { call, put } from 'redux-saga/effects';
+import { getInstance as activeInstance } from './apiCalls';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/sagas/getInstance.js:2]: Why are you doing that?

File: integration-fe/src/sagas/getRooms.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-debugger */
+import { call, put } from 'redux-saga/effects';
+import { getRooms as getUserRooms } from './apiCalls';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/sagas/getRooms.js:3]: Why are you doing that?

File: integration-fe/src/sagas/getUserId.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-debugger */
+import { call, put } from 'redux-saga/effects';
+import { getUserId as getUser } from './apiCalls';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/sagas/getUserId.js:3]: Why are you doing that?

File: integration-fe/src/views/Home.jsx.orig [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (cmarcondes-daitan) [inline integration-fe/src/views/Home.jsx.orig]: This file should be removed.

**Review by cmarcondes-daitan — CHANGES_REQUESTED** [review]

File: integration-fe/src/js/app.jsx (hunk 1 above)
Comment (pdarde-daitan) [reply integration-fe/src/js/app.jsx]: Not really in the app.js. Only in the webpack config file. I'll remove this.

File: integration-fe/src/sagas/getAppName.js (hunk 2 above)
Comment (pdarde-daitan) [reply integration-fe/src/sagas/getAppName.js:3]: This is part of react-saga. The "call" effect calls an api end point, or an api service. If the call executes an async call, it pauses the execution until the async operation has finished.
After the "call" to be completed, we dispatch an action with the "put" effect.

File: integration-fe/src/sagas/getBaseWebHookURL.js (hunk 3 above)
Comment (pdarde-daitan) [reply integration-fe/src/sagas/getBaseWebHookURL.js:3]: The same as the above. Each "saga" file has its own role. This returns the base webhook url from the url params.

File: integration-fe/src/sagas/getInstance.js (hunk 4 above)
Comment (pdarde-daitan) [reply integration-fe/src/sagas/getInstance.js:2]: this saga gets the "active instance". This handles with an in-memory object to create, edit or remove an instance.

File: integration-fe/src/sagas/getRooms.js (hunk 5 above)
Comment (pdarde-daitan) [reply integration-fe/src/sagas/getRooms.js:3]: This saga executes an async operation in order to fetch the user rooms. After fetch, this dispatch an action to update the application state.

File: integration-fe/src/sagas/getUserId.js (hunk 6 above)
Comment (pdarde-daitan) [reply integration-fe/src/sagas/getUserId.js:3]: The same as above. This time, this fetches the user Id.

File: integration-fe/src/views/Home.jsx.orig (hunk 7 above)
Comment (pdarde-daitan) [reply integration-fe/src/views/Home.jsx.orig]: Ok.

**Review by cmarcondes-daitan — APPROVED** [review]

**Pull request merged**